### PR TITLE
refactor(rules): unwrapped module layout

### PR DESCRIPTION
**Pull request by rgrinberg**
Add type safety to the representation of (wrapped false) in the module
layout

Such a layout differs from (wrapped false) in that it never has top
level module alias or toplevel interface module

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

**Comment by rgrinberg** [comment]
I ended up going with the type safe option in the end. It required a little too much explanation in the code.

File: src/dune_rules/modules.ml [hunk 1]
```diff
@@ -447,75 +443,148 @@ module Group = struct
...
+    match Module_name.Map.find t name with
+    | Some (Module m) -> Some m
+    | Some (Group _) | None -> None
 
-  let encode t = Group.encode t
+  let parents t m = Group.parents_modules [] t m
+
+  let find_dep t ~of_ name =
+    match Module.kind of_ with
+    | Alias _ -> Ok []
```
Comment (anmonteiro) [inline src/dune_rules/modules.ml:550]: is this ever the case in `(wrapped false)`? I thought there was no alias module.

Comment (rgrinberg) [reply src/dune_rules/modules.ml:550]: There are alias modules in `(wrapped false)` whenever `(include_subdirs qualified)` is enabled for modules in sub directories. There's just no alias module in the top level.

**Review by anmonteiro — APPROVED** [review]
nice cleanup

**Comment by anmonteiro** [comment]
I feel like this reads a lot better even if there are some duplicated functions. They're mostly 1-2 lines long, though, and it's easy to understand what's going on.

**Review by Alizter — APPROVED** [review]

**Pull request merged**